dspy/modeling/backends/json.py

```diff
@@ -0,0 +1,109 @@
+import json
+import logging
+import typing as t
+from json import JSONDecodeError
+
+from dspy.primitives.example import Example
+from dspy.primitives.prediction import Completions
+from dspy.signatures.signature import Signature
+
+from .text import TextBackend
+
+logger = logging.getLogger(__name__)
+
+
+def patch_example(example: Example, data: dict[str, t.Any]) -> Example:
+    example = example.copy()
+    for k, v in data.items():
+        example[k.lower()] = v
+
+    return example
+
+
+class JSONBackend(TextBackend):
+    def _guidelines(self, signature: Signature, example: Example) -> str:
+        included_fields = []
+        missing_fields = []
+        for name, field in signature.fields.items():
+            if name in example:
+                included_fields.append((name, field))
+            else:
+                missing_fields.append((name, field))
+
+        result = "Provided the following:"
```

Comment:
this is a bit awkward. Can we rephrase it to what we have as "Given the following fields:"